django_spanner/base.py

```diff
@@ -124,7 +127,9 @@ def get_new_connection(self, conn_params):
         return Database.connect(**conn_params)
 
     def init_connection_state(self):
-        pass
+        self.connection.close()
```

Comment:
Won't this create the new connection and then immediately close it? https://github.com/django/django/blob/8093aaa8ff9dd7386a069c6eb49fcc1c5980c033/django/db/backends/base/base.py#L195-L197

Reply:
There is no `connection` property hence no active getter to be called, so `self.connection.close()` directly invokes the `Connection.close()` method that simply rolls back the outstanding transaction, if any, and sets the `is_closed` flag to `True` without destroying the connection object itself. Then we invoke the `__init__()` method that initializes the connection state.

Answering the question, no, this will not create a new connection (in its current implementation) and immediately close it, but rather reset the existing connection to its initial state.